src/lib/Shapes.ts

```diff
@@ -81,8 +81,8 @@ export class Shapes {
 		return new InstanceValidator(expected);
 	}
 
-	public union<T>(...validators: readonly BaseValidator<T>[]) {
-		return new UnionValidator(validators);
+	public union<T extends BaseValidator<unknown>>(validators: readonly T[]): T {
```

Comment:
This return type is wrong, it doesn't return T, it returns an UnionValidator with the unwrapped type.

Also the rest params are intentional, if you use an array, just spread it.

Reply:
Ok then readme needs a fix.